charmcraft/utils.py

```diff
@@ -185,3 +186,26 @@ def get_host_architecture():
     """Get host architecture in deb format suitable for base definition."""
     os_platform = get_os_platform()
     return ARCH_TRANSLATIONS.get(os_platform.machine, os_platform.machine)
+
+
+def confirm_with_user(prompt, default=False) -> bool:
+    """Query user for yes/no answer.
+
+    If TTY is not interaction, returns default value.
+    If answer is empty (""), returns default value.
+
+    :returns: True if yes (Y,y,YES,yes), otherwise false.
+    """
+    if not sys.stdin.isatty():
+        return default
+
+    choices = " [Y/n]: " if default else " [y/N]: "
+    default_answer = "y" if default else "n"
```

Comment:
This looks a bit confusing, maybe refactor to test once for `default`?

Reply:
Agreed, removed and simplified code a bit.